core/src/main/java/org/mskcc/cbio/oncokb/bo/impl/AlterationBoImpl.java

```diff
@@ -178,17 +179,27 @@ public Alteration findAlteration(Gene gene, AlterationType alterationType, Strin
             if (alteration.getAlteration().toLowerCase().equals("translocation")
                 || alteration.getAlteration().toLowerCase().equals("inversion")) {
                 // If no fusions curated, check the Truncating Mutations.
-                if (truncatingMutation != null) {
+                if (truncatingMutation != null && !alterations.contains(truncatingMutation)) {
+                    alterations.add(truncatingMutation);
+                }
+            }
+
+            // Map intragenic to Deletion or Truncating Mutation
+            if (alteration.getAlteration().toLowerCase().contains("intragenic")) {
+                if (deletion != null) {
```

Comment:
Because for this case, if deletion is not null, you don't want to include Truncating Mutations.